Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/GameAdapter.kt

```diff
@@ -65,7 +58,32 @@ class GameAdapter(activity: Activity) : RecyclerView.Adapter<GameViewHolder>(),
         val context = holder.itemView.context
         val gameFile = mGameFiles[position]
 
-        GlideUtils.loadGameCover(holder, holder.binding.imageGameScreen, gameFile, mActivity)
+        holder.apply {
+            if (BooleanSetting.MAIN_SHOW_GAME_TITLES.booleanGlobal) {
+                binding.textGameTitle.text = gameFile.title
+                binding.textGameTitle.visibility = View.VISIBLE
+                binding.textGameTitleInner.visibility = View.GONE
+                binding.textGameCaption.visibility = View.VISIBLE
+            } else {
+                binding.textGameTitleInner.text = gameFile.title
+                binding.textGameTitle.visibility = View.GONE
+                binding.textGameCaption.visibility = View.GONE
```

Comment:
You should probably explicitly set `binding.textGameTitleInner.visibility` to `View.VISIBLE` here, in case we're recycling a view that had it set to `View.GONE`.

Reply:
Good catch